contracts/src/v0.8/dev/UpkeepTranscoder3_0.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./../interfaces/UpkeepTranscoderInterface.sol";
+import "./../interfaces/TypeAndVersionInterface.sol";
+import "./../UpkeepFormat.sol";
+
+/**
+ * @notice Transcoder 3_0 allows converting upkeep data from previous keeper registry versions to registry 2.0
+ */
+contract UpkeepTranscoder3_0 is UpkeepTranscoderInterface, TypeAndVersionInterface {
+  // 1.2
+  struct UpkeepV1 {
+    uint96 balance;
+    address lastKeeper; // 1 storage slot full
+    uint32 executeGas;
+    uint64 maxValidBlocknumber;
+    address target; // 2 storage slots full
+    uint96 amountSpent;
+    address admin; // 3 storage slots full
+  }
```

Comment:
can you import these? (import Upkeep as UpkeepV1 from '..Interface.sol'

Reply:
imported V2 and V3. V1 is defined on `KeeperRegistry1_2.sol` so it cannot be imported.